src/commands.ts

```diff
@@ -22,9 +23,34 @@ export function init(context: ExtensionContext) {
     commands.registerCommand(
       'diffMerge.compareFileWithClipboard',
       compareFileWithClipboard
-    )
+    ),
+    commands.registerCommand('diffMerge.openWithDiffMerge', reopenCurrentWithDiffMerge),
   );
 
+  async function reopenCurrentWithDiffMerge() {
+    const editor = window.activeTextEditor;
+    if (!editor) return;
+    const visibleEditors = window.visibleTextEditors;
+    const editorIndex = visibleEditors.indexOf(editor);
+    // This might not be always true. But it's the only way I can think of how to know whether it's a diff-editor.
+    const isDiffEditor = (editor: TextEditor) => editor.viewColumn === undefined && ['git', 'file'].includes(editor.document.uri.scheme);
+    // In case multiple diff-editors are open, we need to find the correct one.
+    const diffEditors =
+      takeRightWhile(visibleEditors.slice(undefined, editorIndex + 1), isDiffEditor)
+        .concat(takeWhile(visibleEditors.slice(editorIndex + 1), isDiffEditor));
```

Comment:
I wonder, can't we use a regular filter here? e.g.

```suggestion
    const diffEditors = visibleEditors.filter(isDiffEditor);
```

Reply:
What I need here is to find 2 consecutive editors that are diff editors, and then I have to know which one is the left and which one is the right. So, takewhile is used here to take only consecutive ones. filter would take all of them.
The reason I'm using it, is that in some cases you'd have multiple consecutive editors, and you wouldn't know which one is the match for the current editor, e.g. you have 2 diff editors, and yours is the 3rd one, so its match will be the 4th one. Without finding all the consecutive editors, you wouldn't know which group of diff editors it is in.
so yes, filter would give you all the editors that might be a diff editor (using the predicate I defined), and then you can apply the same logic below to find the matching one. but, since this is not a real api to know whether it's a diff editor, it's just a heuristic, I like to relay it to minimum. consider a case there's an editor that appears to be a diff editor by the predicate, if I take only consecutive ones, it would be rare to consider it as a diff editor, as it would happen only if it's a consecutive to the real diff editors.